src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -483,7 +483,31 @@ def _train_epoch(self, context: PhaseContext, silent_mode: bool = False) -> tupl
 
             context.update_context(loss_avg_meter=loss_avg_meter, metrics_compute_fn=self.train_metrics)
 
-            for batch_idx, batch_items in enumerate(progress_bar_train_loader):
+            class PrefetchIterator:
```

Comment:
Very nice trick!
Could you please move this class somewhere outside the Trainer and use it in both train & validation loops?